file.go

```diff
@@ -838,7 +841,30 @@ func (f *FilePages) ReadPage() (Page, error) {
 
 		f.index++
 		if f.skip == 0 {
-			return page, nil
+			// f.skip==0 can be true:
+			//  (1) while reading a row of a column which has multiple values (ie. X.list.element) and values continue
+			//  across pages. In that case we just want to keep reading without skipping any values.
+			//  (2) when seeking to a specific row and trying to reach the start offset of the first
+			//  row in a new page.
+			if !seekToRowStart || header.Type != format.DataPage {
+				// keep reading values from beginning of new page
+				return page, nil
+			}
+			// We need to seek to beginning of row.
+			// V1 data pages do not necessarily start at a row boundary.
+			if page.NumRows() == 0 {
```

Comment:
this is a behavior change and potentially a bug? if we're not seeking it prevents us from returning a page with no rows which may occur on v1 pages? (i'm not sure)

this is safe to remove though. if we are seeking f.skip will never be 0 on a page with 0 rows b/c of the way we subtract numRows here:

https://github.com/parquet-go/parquet-go/blob/main/file.go#L846-L857

Thank you for this work! I think it's quite close.

Reply:
If we are not seeking to the start of a row, then the [previous condition](https://github.com/parquet-go/parquet-go/pull/277/files#diff-33f8fbc61d6878200e61e69725cd724475a0c9dbb73bc63fcfc913a5651764feR849) is `true`, we have already returned the current `page` and this code won't be executed. But please let me know if I maybe misunderstood your comment.

Reply:
my bad! you are correct. i misread this code 🙏